pages/dao/[symbol]/proposal/components/instructions/UXDProtocol/EditIdentityDepository.tsx

```diff
@@ -0,0 +1,206 @@
+import { useContext, useEffect, useState } from 'react'
+import * as yup from 'yup'
+import {
+  Governance,
+  ProgramAccount,
+  serializeInstructionToBase64,
+} from '@solana/spl-governance'
+import {
+  Controller,
+  IdentityDepository,
+  USDC,
+  USDC_DECIMALS,
+  UXD_DECIMALS,
+} from '@uxd-protocol/uxd-client'
+import Input from '@components/inputs/Input'
+import Switch from '@components/Switch'
+import useGovernanceAssets from '@hooks/useGovernanceAssets'
+import { uxdClient } from '@tools/sdk/uxdProtocol/uxdClient'
+import { isFormValid } from '@utils/formValidation'
+import {
+  UiInstruction,
+  UXDEditIdentityDepositoryForm,
+} from '@utils/uiTypes/proposalCreationTypes'
+import { NewProposalContext } from '../../../new'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+import { useConnection } from '@solana/wallet-adapter-react'
+
+const schema = yup.object().shape({
+  governedAccount: yup
+    .object()
+    .nullable()
+    .required('Governance account is required'),
+  mintingDisabled: yup.boolean(),
+  redeemableAmountUnderManagementCap: yup
+    .number()
+    .min(0, 'Redeemable amount under management cap should be min 0'),
+})
+
+const EditIdentityDepository = ({
+  index,
+  governance,
+}: {
+  index: number
+  governance: ProgramAccount<Governance> | null
+}) => {
+  const [mintingDisabledChange, setMintingDisabledChange] = useState<boolean>(
+    false
+  )
+  const [
+    redeemableAmountUnderManagementCapChange,
+    setRedeemableAmountUnderManagementCapChange,
+  ] = useState<boolean>(false)
+
+  const connection = useConnection()
+  const shouldBeGoverned = !!(index !== 0 && governance)
+  const [formErrors, setFormErrors] = useState({})
+  const { handleSetInstructions } = useContext(NewProposalContext)
+  const { assetAccounts } = useGovernanceAssets()
+
+  const [form, setForm] = useState<UXDEditIdentityDepositoryForm>({
+    governedAccount: undefined,
+    mintingDisabled: false,
+    redeemableAmountUnderManagementCap: 0,
+  })
+
+  const handleSetForm = ({ propertyName, value }) => {
+    setFormErrors({})
+    setForm({ ...form, [propertyName]: value })
+  }
+
+  const validateInstruction = async (): Promise<boolean> => {
+    const { isValid, validationErrors } = await isFormValid(schema, form)
+    setFormErrors(validationErrors)
+    return isValid
+  }
+
+  async function getInstruction(): Promise<UiInstruction> {
+    const isValid = await validateInstruction()
+
+    if (
+      !connection ||
+      !isValid ||
+      !form.governedAccount?.governance?.account.governedAccount
+    ) {
+      return {
+        serializedInstruction: '',
+        isValid: false,
+        governance: form.governedAccount?.governance,
+      }
+    }
+    const uxdProgramId =
+      form.governedAccount?.governance?.account.governedAccount
+    const client = uxdClient(uxdProgramId)
```

Comment:
At first glance, it look like the `client` could be `null` if `uxdProgramId` is incorrect?